src/app/chat/views/user_list.nim

```diff
@@ -70,6 +72,58 @@ QtObject:
       UserListRoles.Identicon.int:"identicon"
     }.toTable
 
+  proc add*(self: UserListView, members: seq[ChatMember]) =
+    # Adding chat members
+    for m in members:
+      let pk = m.id
+      if self.userDetails.hasKey(pk): continue
+
+      var userName: string
+      var alias: string
+      var identicon: string
+      var localName: string
+      
+      if self.status.chat.contacts.hasKey(pk):
+        userName = ens.userNameOrAlias(self.status.chat.contacts[pk])
+        alias = self.status.chat.contacts[pk].alias
+        identicon = self.status.chat.contacts[pk].identicon
+        localName = self.status.chat.contacts[pk].localNickname
+      else:
+        userName = m.username
+        alias = m.username
+        identicon = m.identicon
+        localName = ""
+
+      self.beginInsertRows(newQModelIndex(), self.users.len, self.users.len)
+      self.userDetails[pk] = User(
+          userName: userName,
+          alias: alias,
+          localName: localName,
+          lastSeen: "0",
+          identicon: identicon
+      )
+      self.users.add(pk)
+      self.endInsertRows()
+    
+    # Checking for removed members
+    var toDelete: seq[string]
+    for userPublicKey in self.users:
+      var found = false
+      for m in members:
+        if m.id == userPublicKey: 
+          found = true
+          break
+      if not found:
+        toDelete.add(userPublicKey)
+
+    # Removing deleted members
+    if toDelete.len > 0:
+      self.beginResetModel()
```

Comment:
wondering why not https://doc.qt.io/qt-5/qabstractitemmodel.html#beginRemoveRows?